ddl/db_test.go

```diff
@@ -2360,6 +2360,12 @@ func (s *testDBSuite4) TestChangeColumn(c *C) {
 	s.mustExec(tk, c, "alter table t3 add column a bigint")
 	sql = "alter table t3 change aa a bigint"
 	tk.MustGetErrCode(sql, errno.ErrDupFieldName)
+	// https://github.com/pingcap/tidb/issues/23488
+	s.mustExec(tk, c, "drop table if exists t5")
+	s.mustExec(tk, c, "create table t5 (k varchar(10) primary key, v int)")
+	sql = "alter table t5 change column k k tinytext;"
+	tk.MustGetErrCode(sql, mysql.ErrBlobKeyWithoutLength)
+	tk.MustExec("drop table t5")
```

Comment:
Please add the test that "if there is a prefix index on the varchar column, then we can change it to text".

Reply:
@wjhuang2016 as I am new to go-lang, after I add the test code, I only cd ddl and use `go test` now, but the result seems have some fails, I want to run TestChangeColumn() only, and can you tell me how to run the UT usually when add some test code? Thanks!

Reply:
```bash
go test -v github.com/pingcap/tidb/ddl -check.f TestChangeColumn
```

Reply:
@tangenta Thanks, after I run the UT, I found that you checkin some code yesterday, which pervent the conversion from varchar to non-varchar, which makes my UT code useless but check your scanrio.
And about the test scanrio @wjhuang2016 suggested ("if there is a prefix index on the varchar column, then we can change it to text"), it will pass in MySQL. But will fail in TiDB now with your checkin.
MySQL:
![image](https://user-images.githubusercontent.com/767663/113263234-19fb8b80-9304-11eb-86b7-0e05cd0f5428.png)

TiDB:
![image](https://user-images.githubusercontent.com/767663/113262436-2df2bd80-9303-11eb-82a2-d301cabfd475.png)

And also for #23488 , also TiDB behavior is different from MySQL
MySQL
![image](https://user-images.githubusercontent.com/767663/113263523-7494e780-9304-11eb-8e57-ab69e04a01d7.png)

TiDB
![image](https://user-images.githubusercontent.com/767663/113263673-a4dc8600-9304-11eb-9536-27d5eb603586.png)

Could you please help to check?
Thanks!